test/mocks/http/mocks.h

```diff
@@ -317,6 +338,19 @@ class MockAsyncClientRequest : public AsyncClient::Request {
 
   MockAsyncClient* client_;
 };
+
+class MockAsyncClientStream : public AsyncClient::Stream {
```

Comment:
This mock is not used, removed. What are others? MockAsyncClientStreamCallbacks is used in the AsyncClientImplTest.